### ISSUE TITLE
Question about `train.source`

### ISSUE THREAD

**Issue by lryan599** [hhy-huang/GraphJudge#1]
Thank you for your excellent work!

After reading the paper, I still have an unresolved question and would appreciate your guidance.

Does KASFT see partial triples from the ground truth knowledge graph (KG) in advance, potentially leading to data leakage?

In terms of the code, how is `./datasets/GPT4o_mini_result_{dataset}/train.source` generated, and how does it relate to ground truth KG?

Looking forward to your response!

----

感谢你们的工作！

我阅读了论文后有个疑惑尚未得到解答，还请指教。

KASFT是否提前见到ground truth kg的部分三元组，从而导致数据泄漏的问题？

对应到代码中，就是`./datasets/GPT4o_mini_result_{dataset}/train.source`是如何得到的，它与ground truth kg有何关系？

期待你的回复！

**Comment by hhy-huang** [comment]
Hi, thanks so much for your interest in our work!

For your question about the potential data leakage issue between training and test data, we use the existing train/test/valid datasets in previous published works: [REBEL](https://aclanthology.org/2021.findings-emnlp.204.pdf), [GenWiki](https://aclanthology.org/2020.coling-main.217.pdf), [SCIERC](https://aclanthology.org/D18-1360.pdf). So we assume there is no data leakage issue between the train/test/valid datasets of them.

And our ground truth KGs are the KGs in their test set. Our KGs used for training are the KGs in their training set. Sorry for ignoring to upload the training data in our repo. Now we have uploaded them. Please feel free to check them. You can also obtain the them from their official resources :).

**Comment by ApexGBL** [comment]
Hello,

First of all, thank you for sharing the 2DNesting project.
This is exactly the type of algorithm I have been searching for.

I would like to use 2DNesting in my own C# project.
Is it possible to get a compiled DLL version of the nesting engine (without the UI part)?
Or could you provide some guidance on how to build the project as a DLL (CMake / dependencies etc.)?

Any help would be greatly appreciated.

Thank you very much!
